Add scheduled inbox purge and retention setting

pfefferle opened the pull request:
Introduces a scheduled task to purge old inbox items, similar to the existing outbox purge. Adds an advanced setting for configuring the inbox retention period, including UI and option registration. Ensures site performance by allowing administrators to control how long inbox items are kept.

## Proposed changes:
- Adds a new advanced setting for configuring inbox retention period (default 360 days)
- Implements a daily scheduled task to automatically purge old inbox items
- Provides UI for administrators to control the inbox retention setting

### Other information:

- [x] Have you written new tests for your changes, if applicable?

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* See unittests

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger a GitHub workflow that will create and push the entry into the branch. -->

<!-- Due to org permissions, the job may fail for PRs crated from a fork under GitHub organizations. -->
<!-- In this case, you can create entry manually with `composer changelog:add` and push it into the branch. -->

<!-- If no changelog entry is required for this PR, please add the "Skip Changelog" label. -->

-   [X] Automatically create a changelog entry from the details below.

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [X] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [X] Added - for new features
-   [ ] Changed - for changes in existing functionality
-   [ ] Deprecated - for soon-to-be removed features
-   [ ] Removed - for now removed features
-   [ ] Fixed - for any bug fixes
-   [ ] Security - in case of vulnerabilities

#### Message

Added a scheduled task and setting to automatically purge old inbox items, helping maintain site performance and storage control.

</details>

pfefferle committed:
Add scheduled inbox purge and retention setting

Introduces a scheduled task to purge old inbox items, similar to the existing outbox purge. Adds an advanced setting for configuring the inbox retention period, including UI and option registration. Ensures site performance by allowing administrators to control how long inbox items are kept.

pfefferle committed:
Add tests for Scheduler::purge_inbox method

Introduces unit tests for the Scheduler::purge_inbox method, covering scenarios with more than 200 posts, 200 or fewer posts, and varying the inbox purge days option. These tests ensure correct deletion behavior based on post age and configurable purge thresholds.

matticbot committed:
Add changelog

obenland commented:
Instead of introducing separate options and cron jobs, could this be combined with the Outbox purging?

I noticed the retention period being double that of the Outbox. Is 6 months not enough? I'd expect there to be vastly more posts in the Inbox than the Outbox, especially with Following enabled.

pfefferle commented:
The inbox is what will be used by Apps in the future, so I thought it should be at least a different setting! We can reduce it though.

pfefferle committed:
Merge branch 'trunk' into add/purge-inbox

pfefferle committed:
Merge branch 'trunk' into add/purge-inbox

pfefferle committed:
Reduce default inbox purge days from 360 to 180

Updated the default value for 'activitypub_inbox_purge_days' from 360 to 180 in scheduler logic, settings UI, and documentation. Adjusted related test comments to reflect the new default. This change aims to improve site performance by purging inbox items more frequently.

pfefferle commented:
I think we should keep them separated to be more flexible.

obenland reviewed (commented):
Let's not show a UI for the option and keep it CLI or `wp-admin/options.php` only.

pfefferle committed:
Remove inbox and outbox retention period settings

Eliminated the settings fields and rendering methods for 'Outbox Retention Period' and 'Inbox Retention Period' from the advanced settings. This streamlines the settings page by removing options related to message retention.

pfefferle commented:
@obenland like discussed in slack, I removed the settings for Inbox/Outbox purge.

pfefferle committed:
Merge branch 'trunk' into add/purge-inbox

pfefferle committed:
Add global namespace to WordPress function calls

Prefixed WordPress core function calls with a backslash to ensure the global namespace is used. This helps prevent issues if functions with the same names exist in other namespaces.

obenland commented on includes/class-scheduler.php:
```diff
@@ -321,6 +329,40 @@ public static function purge_outbox() {
+		$date     = new \DateTime( 'now', $timezone );
+
+		$date->sub( \DateInterval::createFromDateString( "$days days" ) );
+
+		$post_ids = \get_posts(
+			array(
+				'post_type'   => Inbox::POST_TYPE,
+				'post_status' => 'any',
+				'fields'      => 'ids',
+				'numberposts' => -1,
+				'date_query'  => array(
+					array(
+						'before' => $date->format( 'Y-m-d' ),
+					),
+				),
+			)
```
```suggestion
		$days     = (int) \get_option( 'activitypub_inbox_purge_days', 180 );
		$post_ids = \get_posts(
			array(
				'post_type'   => Inbox::POST_TYPE,
				'post_status' => 'any',
				'fields'      => 'ids',
				'numberposts' => -1,
				'date_query'  => array(
					array(
						'before' => gmdate( 'Y-m-d', time() - ( $days * DAY_IN_SECONDS ) ),
					),
				),
			)
```
Would this work? Might be a bit simpler.

pfefferle replied:
let's see what the tests say!

obenland replied:
A different level of cowboy coding 😄

obenland reviewed (commented):
I think this all looks good, had just one question:

pfefferle committed:
Update includes/class-scheduler.php

Co-authored-by: Konstantin Obenland <obenland@gmx.de>
